plugins/builds/index.js

```diff
@@ -560,15 +564,55 @@ function parseJobInfo({ joinObj, currentJobName, nextJobName, pipelineId, build
 
 /**
  * Get finished builds in all parent events
- * @param  {Event}      event           Current event
- * @param  {Factory}    eventFactory    Event Factory
- * @return {Promise}                    All finished builds
+ * @param  {Event}      event                   Current event
+ * @param  {Number}     [event.parentEventId]   Parent event ID
+ * @param  {Number}     [event.groupEventId]    Group parent event ID
+ * @param  {Factory}    eventFactory            Event Factory
+ * @return {Promise}                            All finished builds
  */
 async function getFinishedBuilds(event, eventFactory) {
     if (!event.parentEventId) {
         return event.getBuilds();
     }
 
+    // New logic to use groupEventId
+    if (event.groupEventId) {
+        const parentEvents = await eventFactory.list({
+            params: {
+                groupEventId: event.groupEventId
+            }
+        });
+        const builds = await event.getBuilds();
+        let parentBuilds = [].concat(builds);
+
+        await Promise.all(
+            parentEvents.map(async pe => {
+                const eventBuilds = await pe.getBuilds();
```

Comment:
can we do a single query here ? something like
```
SELECT * FROM  builds 
     WHERE build.eventId IN (
         SELECT event.id FROM events WHERE event.id = groupEventId
     )
     GROUP BY build.jobId

```
and maybe limit as well ?
else we are looking at multiple db queries equivalent to number of events here, which might not be that many in numbers 🤔  exploring if we can offload all this logic to DB

cc @parthasl

Reply:
so idea is to add a method in build model to fetch builds  (latest) for jobs for a given groupEventId

Reply:
thinking about using a query like
```
SELECT * FROM (
    SELECT *, RANK() OVER ( PARTITION BY "jobId" ORDER BY "id" DESC ) AS rank
    FROM public.builds WHERE "eventId" IN 
        (SELECT "id" FROM public.events WHERE "groupEventId" = '12345')) AS foo
WHERE rank = 1
```
This should return an _array of builds_ consisting of the latest build for each job that ran in an event that has a matching `groupEventId` 12345

Reply:
https://github.com/screwdriver-cd/models/pull/450